packages/app/src/systems/Account/components/BalanceWidget/BalanceWidget.tsx

```diff
@@ -0,0 +1,152 @@
+import { cssObj } from "@fuel-ui/css";
+import { Avatar, Box, Flex, Icon, IconButton, Text } from "@fuel-ui/react";
+import type { ReactNode } from "react";
+
+import type { Account } from "../../types";
+
+import { BalanceWidgetLoader } from "./BalanceWidgetLoader";
+
+import { shortAddress } from "~/systems/Core";
+
+type BalanceWidgetWrapperProps = {
+  children: ReactNode;
+};
+
+export function BalanceWidgetWrapper({ children }: BalanceWidgetWrapperProps) {
+  return (
+    <Box css={styles.balanceWidgetWrapper}>
+      <Box css={styles.backgroundShadow}>&nbsp;</Box>
+      <Box css={styles.backgroundFront}>&nbsp;</Box>
+      <Flex css={styles.contentWrapper}>{children}</Flex>
+    </Box>
+  );
+}
+
+export type BalanceWidgetProps = {
+  account: Account;
+  isHidden?: boolean;
+};
+
+export function BalanceWidget({ account, isHidden }: BalanceWidgetProps) {
+  async function handleCopy() {
+    await navigator.clipboard.writeText(account.address);
+  }
+
+  return (
+    <BalanceWidgetWrapper>
+      <Flex direction="column" align="center">
+        <Avatar
+          size="lg"
+          name={account.name}
+          /**
+           * TODO: Need to add Avatar.Generated here when it's done on @fuel-ui
+           */
+          src="https://images.unsplash.com/photo-1492633423870-43d1cd2775eb?&w=128&h=128&dpr=2&q=80"
+        />
+        <IconButton
+          size="xs"
+          variant="ghost"
+          color="gray"
+          icon={<Icon icon="CaretDown" color="gray8" />}
+          aria-label="Expand"
+          css={styles.caretDownIcon}
+        />
+      </Flex>
+      <Flex justify="space-between" css={{ flex: "1 0" }}>
+        <Flex direction="column" css={{ marginLeft: 14, alignSelf: "center" }}>
+          <Flex align="center">
+            <Text fontSize="sm" color="gray11" css={{ fontWeight: "bold" }}>
+              {shortAddress(account.address)}
+            </Text>
+            <IconButton
+              onPress={handleCopy}
+              size="xs"
+              variant="link"
+              color="gray"
+              icon={<Icon icon="Copy" color="gray8" size={18} weight="thin" />}
+              aria-label="Copy address"
+              css={{ marginLeft: 4 }}
+            />
+          </Flex>
+          <Text
+            color={isHidden ? "gray10" : "gray12"}
+            fontSize="2xl"
+            css={{ fontWeight: "bold" }}
+          >
+            $&nbsp;{isHidden ? "•••••" : account.balance || "0.00"}
+          </Text>
+        </Flex>
+        <Box css={{ marginRight: 6, marginTop: 8 }}>
+          <IconButton
```

Comment:
Maybe we can convert this into a separate component, `VisibilityButton` or something like this?